### PR TITLE
Add useful initialisers to the catalog

### PR DESCRIPTION
**Pull request by duncangrant**
In order to add initialisers to the catalog I've had to make them non-final and add public no-arg constructors.

### CODE REVIEW

File: core/src/test/java/org/apache/brooklyn/core/test/BrooklynAppLiveTestSupport.java [hunk 1]
```diff
@@ -36,10 +36,17 @@
...
         if (mgmt!=null) {
             app = mgmt.getEntityManager().createEntity(newAppSpec());
         } else {
-            mgmt = new LocalManagementContextForTests(BrooklynProperties.Factory.newDefault());
+
+            LocalManagementContextForTests.Builder builder = LocalManagementContextForTests.builder(true).
+                    setOsgiEnablementAndReuse(true, true);
+            if (true) {
+                builder.useDefaultProperties();
+            }
```
Comment (algairim) [inline core/src/test/java/org/apache/brooklyn/core/test/BrooklynAppLiveTestSupport.java:48]: Does not look like `if` statement needed.

Comment (algairim) [reply core/src/test/java/org/apache/brooklyn/core/test/BrooklynAppLiveTestSupport.java:48]: I mean, `if (true)`

**Review by algairim — APPROVED** [review]
One minor comment, apart from that looks good.

**Review by zan-mateusz — APPROVED** [review]
LGTM!

File: core/src/test/java/org/apache/brooklyn/core/test/BrooklynAppLiveTestSupport.java (hunk 1 above)
Comment (duncangrant) [reply core/src/test/java/org/apache/brooklyn/core/test/BrooklynAppLiveTestSupport.java:48]: I'm not sure where this change has come from - I don't have it locally?

Comment (duncangrant) [reply core/src/test/java/org/apache/brooklyn/core/test/BrooklynAppLiveTestSupport.java:48]: I must have messed up a rebase

**Pull request merged**